pyomo/contrib/doe/fim_doe.py

```diff
@@ -1035,17 +1114,30 @@ def __create_doe_model(self):
         # Elements in Jacobian matrix
         if self.jac_initial is not None:
             dict_jac = {}
-            for i, bu in enumerate(m.y_set):
-                for j, un in enumerate(m.para_set):
-                    for t, tim in enumerate(m.tmea_set):
-                        dict_jac[(bu,un,tim)] = self.jac_initial[i,j,t]
+            if jac_involved_name is not None:
```

Comment:
Simplify this? You could instead say:
```
if jac_involved_name is not None:
   measurements = m.involved_y_set
else:
   measurements = m.y_set
```

Then use `measurements` in the for loops. But I do not fully appreciate why this logic is needed.